packages/main/src/MultiComboBox.ts

```diff
@@ -1578,12 +1587,14 @@ class MultiComboBox extends UI5Element implements IFormInputElement {
 		const value = input && input.value;
 
 		if (this.open) {
-			this._getList().then(list => {
-				const selectedItemsCount = list?.querySelectorAll("[ui5-li][selected]")?.length;
-				const allItemsCount = list?.querySelectorAll("[ui5-li]")?.length;
-				this._allSelected = selectedItemsCount === allItemsCount;
-			});
+			const list = this._getList();
+			const selectedListItemsCount = list?.querySelectorAll("[ui5-li][selected]")?.length;
+			const items = this._getItems();
+			const listItems = list?.querySelectorAll("[ui5-li]")?.length;
+			const selectedItemsCount = items.filter(item => item.selected).length;
+			this._allSelected = selectedItemsCount === items.length || selectedListItemsCount === listItems;
```

Comment:
this._allSelected = selectedItemsCount === items.length || selectedListItemsCount === listItems;
is this correct for the group scenario as well? this._getItems() would return the group items as well?

Reply:
Yes, this results in a bug.